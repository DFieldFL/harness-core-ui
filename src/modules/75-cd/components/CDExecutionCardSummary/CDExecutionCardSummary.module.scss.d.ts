/* eslint-disable */
// this is an auto-generated file
declare const styles: {
  readonly cardSummary: string
  readonly heading: string
  readonly servicesContainer: string
  readonly servicesIcon: string
  readonly servicesList: string
  readonly serviceWrapper: string
  readonly serviceName: string
  readonly toggle: string
}
export default styles
