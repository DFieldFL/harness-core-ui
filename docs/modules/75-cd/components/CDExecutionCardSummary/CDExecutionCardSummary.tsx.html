
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for modules/75-cd/components/CDExecutionCardSummary/CDExecutionCardSummary.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../../index.html">All files</a> / <a href="index.html">modules/75-cd/components/CDExecutionCardSummary</a> CDExecutionCardSummary.tsx</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">82.35% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>28/34</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">62.16% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>23/37</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">57.14% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>4/7</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">87.5% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>28/32</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a>
<a name='L74'></a><a href='#L74'>74</a>
<a name='L75'></a><a href='#L75'>75</a>
<a name='L76'></a><a href='#L76'>76</a>
<a name='L77'></a><a href='#L77'>77</a>
<a name='L78'></a><a href='#L78'>78</a>
<a name='L79'></a><a href='#L79'>79</a>
<a name='L80'></a><a href='#L80'>80</a>
<a name='L81'></a><a href='#L81'>81</a>
<a name='L82'></a><a href='#L82'>82</a>
<a name='L83'></a><a href='#L83'>83</a>
<a name='L84'></a><a href='#L84'>84</a>
<a name='L85'></a><a href='#L85'>85</a>
<a name='L86'></a><a href='#L86'>86</a>
<a name='L87'></a><a href='#L87'>87</a>
<a name='L88'></a><a href='#L88'>88</a>
<a name='L89'></a><a href='#L89'>89</a>
<a name='L90'></a><a href='#L90'>90</a>
<a name='L91'></a><a href='#L91'>91</a>
<a name='L92'></a><a href='#L92'>92</a>
<a name='L93'></a><a href='#L93'>93</a>
<a name='L94'></a><a href='#L94'>94</a>
<a name='L95'></a><a href='#L95'>95</a>
<a name='L96'></a><a href='#L96'>96</a>
<a name='L97'></a><a href='#L97'>97</a>
<a name='L98'></a><a href='#L98'>98</a>
<a name='L99'></a><a href='#L99'>99</a>
<a name='L100'></a><a href='#L100'>100</a>
<a name='L101'></a><a href='#L101'>101</a>
<a name='L102'></a><a href='#L102'>102</a>
<a name='L103'></a><a href='#L103'>103</a>
<a name='L104'></a><a href='#L104'>104</a></td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import { Icon, Popover } from '@wings-software/uicore'
import { Position } from '@blueprintjs/core'
&nbsp;
import { String } from 'framework/strings'
import type { ExecutionCardInfoProps } from '@pipeline/factories/ExecutionFactory/types'
import type { CDPipelineModuleInfo, CDStageModuleInfo, ServiceExecutionSummary } from 'services/cd-ng'
import { getPipelineStagesMap } from '@pipeline/utils/executionUtils'
import { ServicePopoverCard } from '@cd/components/ServicePopoverCard/ServicePopoverCard'
import { CardVariant } from '@pipeline/utils/constants'
&nbsp;
import css from './CDExecutionCardSummary.module.scss'
&nbsp;
const SERVICES_LIMIT = 5
&nbsp;
export function CDExecutionCardSummary(props: ExecutionCardInfoProps): React.ReactElement {
  const { data, nodeMap, startingNodeId, variant } = props
  const serviceIdentifiers: string[] = ((data as CDPipelineModuleInfo)<span class="branch-0 cbranch-no" title="branch not covered" >?.s</span>erviceIdentifiers as string[]) || <span class="branch-1 cbranch-no" title="branch not covered" >[]</span>
  const [showMore, setShowMore] = React.useState(false)
  const servicesMap = React.useMemo(() =&gt; {
    const stagesMap = getPipelineStagesMap(nodeMap, startingNodeId)
    const map = new Map&lt;string, ServiceExecutionSummary&gt;()
&nbsp;
    stagesMap.forEach(stage =&gt; {
      const serviceInfo = (stage.moduleInfo<span class="branch-0 cbranch-no" title="branch not covered" >?.c</span>d as CDStageModuleInfo)<span class="branch-0 cbranch-no" title="branch not covered" >?.s</span>erviceInfo
&nbsp;
      // istanbul ignore else
      if (serviceInfo<span class="branch-0 cbranch-no" title="branch not covered" >?.i</span>dentifier) {
        map.set(serviceInfo.identifier, serviceInfo)
      }
    })
&nbsp;
    return map
  }, [nodeMap, startingNodeId])
  const hasMoreItems = serviceIdentifiers.length &gt; SERVICES_LIMIT
  const items = showMore &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >hasMoreItems </span>? <span class="branch-0 cbranch-no" title="branch not covered" >serviceIdentifiers </span>: serviceIdentifiers<span class="branch-0 cbranch-no" title="branch not covered" >?.s</span>lice(0, SERVICES_LIMIT)
&nbsp;
  function <span class="fstat-no" title="function not covered" >toggleSection(</span>e: React.SyntheticEvent): void {
<span class="cstat-no" title="statement not covered" >    e.stopPropagation()</span>
<span class="cstat-no" title="statement not covered" >    setShowMore(<span class="fstat-no" title="function not covered" >status </span>=&gt; <span class="cstat-no" title="statement not covered" >!status)</span></span>
  }
&nbsp;
  function <span class="fstat-no" title="function not covered" >killEvent(</span>e: React.SyntheticEvent): void {
<span class="cstat-no" title="statement not covered" >    e.stopPropagation()</span>
  }
&nbsp;
  return (
    &lt;div className={css.cardSummary}&gt;
      &lt;String
        tagName="div"
        className={css.heading}
        stringID="executionList.servicesDeployedText"
        vars={{ size: serviceIdentifiers.length }}
      /&gt;
      {serviceIdentifiers.length &gt; 0 ? (
        &lt;div className={css.servicesContainer}&gt;
          &lt;Icon name="services" className={css.servicesIcon} size={16} /&gt;
          &lt;div className={css.servicesList}&gt;
            {items.map((identifier: string) =&gt; {
              <span class="missing-if-branch" title="else path not taken" >E</span>if (variant === CardVariant.Default) {
                const service = servicesMap.get(identifier)
&nbsp;
                <span class="missing-if-branch" title="if path not taken" >I</span>if (!service) <span class="cstat-no" title="statement not covered" >return null</span>
&nbsp;
                return (
                  &lt;Popover
                    key={identifier}
                    wrapperTagName="div"
                    targetTagName="div"
                    interactionKind="hover"
                    position={Position.BOTTOM_RIGHT}
                    className={css.serviceWrapper}
                  &gt;
                    &lt;div className={css.serviceName} onClick={killEvent}&gt;
                      {service.displayName}
                    &lt;/div&gt;
                    &lt;ServicePopoverCard service={service} /&gt;
                  &lt;/Popover&gt;
                )
              } else {
<span class="cstat-no" title="statement not covered" >                return (</span>
                  &lt;div className={css.serviceWrapper}&gt;
                    &lt;div key={identifier} className={css.serviceName}&gt;
                      {identifier}
                    &lt;/div&gt;
                  &lt;/div&gt;
                )
              }
            })}
          &lt;/div&gt;
          {hasMoreItems ? (
<span class="branch-0 cbranch-no" title="branch not covered" >            &lt;String</span>
              className={css.toggle}
              onClick={toggleSection}
              stringID={showMore ? 'common.showLess' : 'common.plusNumberNoSpace'}
              vars={{ number: Math.abs(serviceIdentifiers.length - SERVICES_LIMIT) }}
            /&gt;
          ) : null}
        &lt;/div&gt;
      ) : <span class="branch-1 cbranch-no" title="branch not covered" >null}</span>
    &lt;/div&gt;
  )
}
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Tue Oct 05 2021 04:31:17 GMT+0000 (Greenwich Mean Time)
            </div>
        </div>
        <script src="../../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../../sorter.js"></script>
        <script src="../../../../block-navigation.js"></script>
    </body>
</html>
    