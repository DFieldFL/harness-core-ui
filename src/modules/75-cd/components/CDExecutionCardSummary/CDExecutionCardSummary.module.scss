.cardSummary {
  display: flex;
  flex-direction: column;
  row-gap: var(--spacing-3);

  .heading {
    font-size: var(--font-size-small);
    font-weight: bold;
    color: var(--grey-600);
  }

  .servicesContainer {
    position: relative;
    padding-left: var(--spacing-7);

    .servicesIcon {
      position: absolute;
      left: 0;
      top: 2px;
      transform: scale(1.5) translateX(-3px);
      transform-origin: left center;
    }

    .servicesList {
      display: inline-flex;
      align-items: center;
    }

    .serviceWrapper {
      display: inline-flex;
      align-items: center;

      &:not(:last-child)::after {
        content: ',';
        color: var(--primary-7);
        margin-right: 0.33em;
      }

      :global(.bp3-popover-target) {
        display: inline-block;
      }
    }

    .serviceName {
      color: var(--primary-7);
      font-size: var(--font-size-small);
    }
  }

  .toggle {
    color: var(--primary-7);
    font-size: var(--font-size-small);
    text-decoration: underline;
    cursor: pointer;
    display: inline-block;
    line-height: 18px;

    &::before {
      content: ',';
      color: var(--primary-7);
      margin-right: 0.33em;
    }
  }
}
