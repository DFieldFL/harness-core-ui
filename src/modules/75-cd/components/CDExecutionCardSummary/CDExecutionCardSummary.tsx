import React from 'react'
import { Icon, Popover } from '@wings-software/uicore'
import { Position } from '@blueprintjs/core'

import { String } from 'framework/strings'
import type { ExecutionCardInfoProps } from '@pipeline/factories/ExecutionFactory/types'
import type { CDPipelineModuleInfo, CDStageModuleInfo, ServiceExecutionSummary } from 'services/cd-ng'
import { getPipelineStagesMap } from '@pipeline/utils/executionUtils'
import { ServicePopoverCard } from '@cd/components/ServicePopoverCard/ServicePopoverCard'

import css from './CDExecutionCardSummary.module.scss'

const SERVICES_LIMIT = 5

export function CDExecutionCardSummary(props: ExecutionCardInfoProps): React.ReactElement {
  const { data, nodeMap, startingNodeId } = props
  const serviceIdentifiers: string[] = ((data as CDPipelineModuleInfo)?.serviceIdentifiers as string[]) || []
  const [showMore, setShowMore] = React.useState(false)
  const servicesMap = React.useMemo(() => {
    const stagesMap = getPipelineStagesMap(nodeMap, startingNodeId)
    const map = new Map<string, ServiceExecutionSummary>()

    stagesMap.forEach(stage => {
      const serviceInfo = (stage.moduleInfo?.cd as CDStageModuleInfo)?.serviceInfo

      // istanbul ignore else
      if (serviceInfo?.identifier) {
        map.set(serviceInfo.identifier, serviceInfo)
      }
    })

    return map
  }, [nodeMap, startingNodeId])
  const hasMoreItems = serviceIdentifiers.length > SERVICES_LIMIT
  const items = showMore && hasMoreItems ? serviceIdentifiers : serviceIdentifiers?.slice(0, SERVICES_LIMIT)

  function toggleSection(e: React.SyntheticEvent): void {
    e.stopPropagation()
    setShowMore(status => !status)
  }

  function killEvent(e: React.SyntheticEvent): void {
    e.stopPropagation()
  }

  return (
    <div className={css.cardSummary}>
      <String
        tagName="div"
        className={css.heading}
        stringID="executionList.servicesDeployedText"
        vars={{ size: serviceIdentifiers.length }}
      />
      {serviceIdentifiers.length > 0 ? (
        <div className={css.servicesContainer}>
          <Icon name="services" className={css.servicesIcon} size={16} />
          <div className={css.servicesList}>
            {items.map((identifier: string) => {
              const service = servicesMap.get(identifier)

              if (!service) return null

              return (
                <Popover
                  key={identifier}
                  wrapperTagName="div"
                  targetTagName="div"
                  interactionKind="hover"
                  position={Position.BOTTOM_RIGHT}
                  className={css.serviceWrapper}
                >
                  <div className={css.serviceName} onClick={killEvent}>
                    {service.displayName}
                  </div>
                  <ServicePopoverCard service={service} />
                </Popover>
              )
            })}
          </div>
          {hasMoreItems ? (
            <String
              className={css.toggle}
              onClick={toggleSection}
              stringID={showMore ? 'common.showLess' : 'common.plusNumberNoSpace'}
              vars={{ number: Math.abs(serviceIdentifiers.length - SERVICES_LIMIT) }}
            />
          ) : null}
        </div>
      ) : null}
    </div>
  )
}
