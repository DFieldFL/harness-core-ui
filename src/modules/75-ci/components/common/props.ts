export type DivAttributesProps = React.HTMLAttributes<HTMLDivElement>

export type ButtonAttributesProps = React.HTMLAttributes<HTMLButtonElement>
