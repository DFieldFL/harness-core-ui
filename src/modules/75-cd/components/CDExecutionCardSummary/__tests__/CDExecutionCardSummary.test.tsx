import React from 'react'
import { render } from '@testing-library/react'

import { CDExecutionCardSummary } from '../CDExecutionCardSummary'

import props from './props.json'

describe('<CDExecutionCardSummary /> tests', () => {
  test('snapshot test', () => {
    const { container } = render(<CDExecutionCardSummary {...(props as any)} />)
    expect(container).toMatchSnapshot()
  })
})
