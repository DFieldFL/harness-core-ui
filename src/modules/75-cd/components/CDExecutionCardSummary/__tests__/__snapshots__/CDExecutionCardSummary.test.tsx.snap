// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CDExecutionCardSummary /> tests snapshot test 1`] = `
<div>
  <div
    class="cardSummary"
  >
    <div>
      No valid template with id "executionList.servicesDeployedText" found in any namespace
    </div>
    <div
      class="servicesContainer"
    >
      <span
        class="bp3-icon StyledProps--main servicesIcon"
        data-icon="services"
      >
        <svg
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M6.74 2.098a.6.6 0 01.6-.6h1.2a.6.6 0 01.6.6v1.4a4.57 4.57 0 011.083.448l.996-.993a.6.6 0 01.847 0l.85.847a.6.6 0 010 .85l-.993.99c.191.333.342.692.445 1.071h1.408a.6.6 0 01.6.6v1.2a.6.6 0 01-.6.6h-1.408a4.54 4.54 0 01-.447 1.073l.994.99a.6.6 0 010 .85l-.85.846a.6.6 0 01-.848 0l-.996-.992a4.574 4.574 0 01-1.08.445v1.401a.6.6 0 01-.6.6h-1.2a.6.6 0 01-.6-.6v-1.4a4.573 4.573 0 01-1.082-.447l-.996.992a.6.6 0 01-.847 0l-.85-.846a.6.6 0 010-.85l.993-.99a4.536 4.536 0 01-.447-1.072H2.106a.6.6 0 01-.6-.6v-1.2a.6.6 0 01.6-.6h1.406a4.52 4.52 0 01.446-1.072l-.993-.989a.6.6 0 010-.85l.85-.847a.6.6 0 01.847 0l.996.992c.336-.192.7-.343 1.083-.446V2.098zm.8.2v1.353a.607.607 0 01-.47.59c-.402.094-.78.252-1.122.462a.607.607 0 01-.747-.084L4.24 3.66l-.567.565.96.956a.607.607 0 01.084.75c-.21.339-.368.713-.463 1.111a.607.607 0 01-.588.47h-1.36v.8h1.36c.29 0 .526.204.588.468.095.399.253.773.463 1.112a.607.607 0 01-.084.75l-.96.955.567.565.963-.959a.607.607 0 01.747-.084c.341.21.719.369 1.121.463.265.061.47.297.47.589v1.353h.8v-1.353a.61.61 0 01.47-.59 3.773 3.773 0 001.121-.462.607.607 0 01.747.085l.963.959.567-.565-.96-.955a.607.607 0 01-.084-.75c.21-.34.369-.714.463-1.113a.607.607 0 01.59-.469h1.359v-.8h-1.36a.607.607 0 01-.589-.469 3.735 3.735 0 00-.462-1.11.607.607 0 01.085-.75l.96-.957-.567-.564-.963.959a.607.607 0 01-.747.084A3.773 3.773 0 008.81 4.24a.607.607 0 01-.47-.59V2.298h-.8zM8.016 6.1a1.9 1.9 0 100 3.8 1.9 1.9 0 000-3.8zM5.315 8a2.7 2.7 0 115.4 0 2.7 2.7 0 01-5.4 0z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <div
        class="servicesList"
      >
        <div
          class="bp3-popover-wrapper serviceWrapper"
        >
          <div
            class="bp3-popover-target"
          >
            <div
              class="serviceName"
              tabindex="0"
            >
              todolist
            </div>
          </div>
        </div>
        <div
          class="bp3-popover-wrapper serviceWrapper"
        >
          <div
            class="bp3-popover-target"
          >
            <div
              class="serviceName"
              tabindex="0"
            >
              testsvc_1
            </div>
          </div>
        </div>
        <div
          class="bp3-popover-wrapper serviceWrapper"
        >
          <div
            class="bp3-popover-target"
          >
            <div
              class="serviceName"
              tabindex="0"
            >
              service1
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
